### PR TITLE
MINOR: Adding system test for named repartition topics

### PR DESCRIPTION
**Pull request by bbejeck**
This is a system test for doing a rolling upgrade of a topology with a named repartition topic.

1. An initial Kafka Streams application is started on 3 nodes.  The topology has one operation forcing a repartition and the repartition topic is explicitly named.
2. Each node is started and processing of data is validated
3. Then one node is stopped (full stop is verified)
4. A property is set signaling the node to add operations to the topology _**before**_ the repartition node which forces a renumbering of all operators (except repartition node)
5. Restart the node and confirm processing records
6. Repeat the steps for the other 2 nodes completing the rolling upgrade 

I ran two runs of the system test with 25 repeats in each run for a total of 50 test runs.
All test runs passed

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by bbejeck** [comment]
ping @guozhangwang, @mjsax, and @vvcephei for reviews

File: streams/src/test/java/org/apache/kafka/streams/tests/StreamsNamedRepartitionTest.java
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/tests/StreamsNamedRepartitionTest.java]: nit: `StreamsUpgradeTest` -> `StreamsNamedRepartitionTest`

File: streams/src/test/java/org/apache/kafka/streams/tests/StreamsNamedRepartitionTest.java
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/tests/StreamsNamedRepartitionTest.java]: nit: simplify to `final String propFileName = args[0];` -- was checked above already

File: streams/src/test/java/org/apache/kafka/streams/tests/StreamsNamedRepartitionTest.java
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/tests/StreamsNamedRepartitionTest.java]: Should we check if all properties are set (ie, `null` check)?

File: streams/src/test/java/org/apache/kafka/streams/tests/StreamsNamedRepartitionTest.java
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/tests/StreamsNamedRepartitionTest.java]: Why do we need this? This is logged anyway.

Comment (bbejeck) [reply streams/src/test/java/org/apache/kafka/streams/tests/StreamsNamedRepartitionTest.java]: nope taking out

File: streams/src/test/java/org/apache/kafka/streams/tests/StreamsNamedRepartitionTest.java
Comment (bbejeck) [reply streams/src/test/java/org/apache/kafka/streams/tests/StreamsNamedRepartitionTest.java]: ack

File: streams/src/test/java/org/apache/kafka/streams/tests/StreamsNamedRepartitionTest.java
Comment (bbejeck) [reply streams/src/test/java/org/apache/kafka/streams/tests/StreamsNamedRepartitionTest.java]: ack

File: streams/src/test/java/org/apache/kafka/streams/tests/StreamsNamedRepartitionTest.java
Comment (bbejeck) [reply streams/src/test/java/org/apache/kafka/streams/tests/StreamsNamedRepartitionTest.java]: ack

**Comment by bbejeck** [comment]
@mjsax, thanks for the review, I've updated per your comments.

**Comment by mjsax** [comment]
Call for second review @guozhangwang @vvcephei

File: streams/src/test/java/org/apache/kafka/streams/tests/StreamsNamedRepartitionTest.java
Comment (vvcephei) [inline streams/src/test/java/org/apache/kafka/streams/tests/StreamsNamedRepartitionTest.java:102]: would it clarify the pattern matching in the ducktape driver if we make this not a prefix of the other case?

File: tests/kafkatest/tests/streams/streams_named_repartition_topic_test.py
Comment (vvcephei) [inline tests/kafkatest/tests/streams/streams_named_repartition_topic_test.py]: is it important to have 3 brokers for this test? I'm wondering if the tests would be more resilient and faster with just one broker and replica of each topic.

File: tests/kafkatest/tests/streams/streams_named_repartition_topic_test.py
Comment (vvcephei) [inline tests/kafkatest/tests/streams/streams_named_repartition_topic_test.py:82]: Sorry for the dumb question... I guess this only tails the file, so there's no danger of a false positive from seeing the patter in the file from before the restart?

**Review by vvcephei — COMMENTED** [review]
Thanks, Bill! It looks good to me. just a couple of comments...

File: streams/src/test/java/org/apache/kafka/streams/tests/StreamsNamedRepartitionTest.java
Comment (bbejeck) [reply streams/src/test/java/org/apache/kafka/streams/tests/StreamsNamedRepartitionTest.java:102]: ack

File: tests/kafkatest/tests/streams/streams_named_repartition_topic_test.py
Comment (bbejeck) [reply tests/kafkatest/tests/streams/streams_named_repartition_topic_test.py]: not sure that would have an impact, but for this test, a single broker is enough, so I'll adjust it.

File: tests/kafkatest/tests/streams/streams_named_repartition_topic_test.py
Comment (bbejeck) [reply tests/kafkatest/tests/streams/streams_named_repartition_topic_test.py:82]: Yes that is correct this method delegates to `LogMonitor#wait_for` which grabs the offset of the file when the object is created and waits for the specified pattern in the file from that point going forward

https://github.com/confluentinc/ducktape/blob/master/ducktape/cluster/remoteaccount.py#L586
https://github.com/confluentinc/ducktape/blob/master/ducktape/cluster/remoteaccount.py#L675

**Comment by bbejeck** [comment]
updated this and rebased from trunk

**Comment by bbejeck** [comment]
kicked off new system test https://jenkins.confluent.io/job/system-test-kafka-branch-builder/2105/

**Comment by mjsax** [comment]
Retest this please.

**Review by mjsax — APPROVED** [review]
LGTM -- assuming Jenkins passes

**Pull request merged**
